tests/integration/embed/embed_test.go

```diff
@@ -206,11 +206,15 @@ func setupEmbedCfg(cfg *embed.Config, curls []url.URL, purls []url.URL) {
 	cfg.ClusterState = "new"
 	cfg.ListenClientUrls, cfg.AdvertiseClientUrls = curls, curls
 	cfg.ListenPeerUrls, cfg.AdvertisePeerUrls = purls, purls
-	cfg.InitialCluster = ""
-	for i := range purls {
-		cfg.InitialCluster += ",default=" + purls[i].String()
+
+	if len(purls) > 0 {
+		builder := new(strings.Builder)
+		builder.WriteString(fmt.Sprintf("default=%s", purls[0].String()))
+		for _, purl := range purls[1:] {
```

Comment:
how about using `"default=" + strings.Join(purls, ",default=")`

Reply:
`purls []url.URL` is not `[]string`
```go
go doc strings.Join

package strings // import "strings"

func Join(elems []string, sep string) string
    Join concatenates the elements of its first argument to create a single
    string. The separator string sep is placed between elements in the resulting
    string.
```

Reply:
oh, I see